src/sources/kubernetes/mod.rs

```diff
@@ -300,6 +267,93 @@ fn transform_file() -> crate::Result<Box<dyn Transform>> {
     })
 }
 
+/// Contains several regexes that can parse common forms of pod_uid.
+/// On first message, regexes are tryed out one after the other until
+/// first succesfull one has been found. After that that regex will be
+/// always used.
+///
+/// If nothing succeds the message is still passed.
+fn transform_pod_uid() -> crate::Result<ApplicableTransform> {
+    let mut regexes = Vec::new();
+
+    let namespace_regex = r"(?P<pod_namespace>[0-9a-z.\-]*)";
+    let name_regex = r"(?P<pod_name>[0-9a-z.\-]*)";
+    let uid_regex = r"(?P<object_uid>([0-9A-Fa-f]{8}[-][0-9A-Fa-f]{4}[-][0-9A-Fa-f]{4}[-][0-9A-Fa-f]{4}[-][0-9A-Fa-f]{12}|[0-9A-Fa-f]{32}))";
+
+    // Definition of pod_uid has been well defined since Kubernetes 1.14 with https://github.com/kubernetes/kubernetes/pull/74441
```

Comment:
is this somewhat new? I thought `pod_uid` _were_ not well defined?

Reply:
Somewhat. It isn't defined by some specification, yet, but has been added/defined with the commented PR. Which I found only recently.

CRI specification will probably be updated in the near future with it. So currently it's a bit of a stretch to call it well defined, but I anticipate that it will.

It actually landed in 1.13, for example GKE 1.13 has it, but others don't. While all tested 1.14 versions contained it.  

EKS also switched recently to higher version, so it's also observed there.

In the end, while I had an idea that platforms are modifying Kubernetes heavily, in actuality,  all tested platforms use main Kubernetes implementation only with slight modifications, if any.

Reply:
cool, we should make sure that this is well documented in our kube guide/docs.